File: VECTOAux/VectoAuxiliaries/Hvac/ISSMCalculate.vb
Namespace Hvac

   Public Interface ISSMCalculate

       Property Run1 As ISSMRun
       Property Run2 As ISSMRun



       'BaseValues
       '- Heating
       ReadOnly Property BaseHeatingW_Mechanical                   As Double 
       ReadOnly Property BaseHeatingW_ElectricalCoolingHeating     As Double 
       ReadOnly Property BaseHeatingW_ElectricalVentilation        As Double 
       ReadOnly Property BaseHeatingW_FuelFiredHeating             As Double
                                                                   
       'Cooling                                                    
       ReadOnly Property BaseCoolingW_Mechanical                   As Double 
       ReadOnly Property BaseCoolingW_ElectricalCoolingHeating     As Double 
       ReadOnly Property BaseCoolingW_ElectricalVentilation        As Double 
       ReadOnly Property BaseCoolingW_FuelFiredHeating            As Double
       
       'Cooling
       ReadOnly Property BaseVentilationW_Mechanical               As Double 
       ReadOnly Property BaseVentilationW_ElectricalCoolingHeating As Double 
       ReadOnly Property BaseVentilationW_ElectricalVentilation    As Double 
       ReadOnly Property BaseVentilationW_FuelFiredHeating        As Double
  
       
       'TechListBenefits
       '- Heating
       ReadOnly Property TechListAdjustedHeatingW_Mechanical                   As Double 
       ReadOnly Property TechListAdjustedHeatingW_ElectricalCoolingHeating     As Double 
       ReadOnly Property TechListAdjustedHeatingW_ElectricalVentilation        As Double 
       ReadOnly Property TechListAdjustedHeatingW_FuelFiredHeating            As Double
                                                                               
       'Cooling          TechListAdjusted                                      
       ReadOnly Property TechListAdjustedCoolingW_Mechanical                   As Double 
       ReadOnly Property TechListAdjustedCoolingW_ElectricalCoolingHeating     As Double 
       ReadOnly Property TechListAdjustedCoolingW_ElectricalVentilation        As Double 
       ReadOnly Property TechListAdjustedCoolingW_FuelFiredHeating            As Double
                        
       'Cooling          TechListAdjusted
       ReadOnly Property TechListAdjustedVentilationW_Mechanical               As Double 
       ReadOnly Property TechListAdjustedVentilationW_ElectricalCoolingHeating As Double 
       ReadOnly Property TechListAdjustedVentilationW_ElectricalVentilation    As Double 
       ReadOnly Property TechListAdjustedVentilationW_FuelFiredHeating        As Double
   
   
   End interface

   

End Namespace

File: VECTOAux/BusAuxiliaries/Interfaces/DownstreamModules/HVAC/ISSMCalculate.cs
namespace TUGraz.VectoCore.BusAuxiliaries.Interfaces.DownstreamModules.HVAC
{
	public interface ISSMCalculate
	{
		ISSMRun Run1 { get; set; }
		ISSMRun Run2 { get; set; }


		double ElectricalWBase { get; }
		double MechanicalWBase { get; }
		double FuelPerHBase { get; }

		double ElectricalWAdjusted { get; }
		double MechanicalWBaseAdjusted { get; }
		double FuelPerHBaseAdjusted { get; }


		// BaseValues
		// - Heating
		double BaseHeatingW_Mechanical { get; }
		double BaseHeatingW_ElectricalCoolingHeating { get; }
		double BaseHeatingW_ElectricalVentilation { get; }
		double BaseHeatingW_FuelFiredHeating { get; }

		// Cooling                                                    
		double BaseCoolingW_Mechanical { get; }
		double BaseCoolingW_ElectricalCoolingHeating { get; }
		double BaseCoolingW_ElectricalVentilation { get; }
		double BaseCoolingW_FuelFiredHeating { get; }

		// Cooling
		double BaseVentilationW_Mechanical { get; }
		double BaseVentilationW_ElectricalCoolingHeating { get; }
		double BaseVentilationW_ElectricalVentilation { get; }
		double BaseVentilationW_FuelFiredHeating { get; }


		// TechListBenefits
		// - Heating
		double TechListAdjustedHeatingW_Mechanical { get; }
		double TechListAdjustedHeatingW_ElectricalCoolingHeating { get; }
		double TechListAdjustedHeatingW_ElectricalVentilation { get; }
		double TechListAdjustedHeatingW_FuelFiredHeating { get; }

		// Cooling          TechListAdjusted                                      
		double TechListAdjustedCoolingW_Mechanical { get; }
		double TechListAdjustedCoolingW_ElectricalCoolingHeating { get; }
		double TechListAdjustedCoolingW_ElectricalVentilation { get; }
		double TechListAdjustedCoolingW_FuelFiredHeating { get; }

		// Cooling          TechListAdjusted
		double TechListAdjustedVentilationW_Mechanical { get; }
		double TechListAdjustedVentilationW_ElectricalCoolingHeating { get; }
		double TechListAdjustedVentilationW_ElectricalVentilation { get; }
		double TechListAdjustedVentilationW_FuelFiredHeating { get; }
	}
}
